share/lib/python/neuron/rxd/rxd.py

```diff
@@ -567,6 +567,8 @@ def _matrix_to_rxd_sparse(m):
 # TODO: make sure this does the right thing when the diffusion constant changes between two neighboring nodes
 def _setup_matrices():
 
+    initializer._init_lock.acquire()
```

Comment:
Just a question (not familiar with the code), would it make sense to have a `try` with a `finally: _init_lock.release()` ? Exception may be raised between `acquire` and `release`

Reply:
That's a good point.

Also: is there any possibility for the `_init_lock` to interfere with the GUI event system? I ask because (1) I believe this sequence of locks got started because of the GUI thread triggering something internal, and (2) it's possible for `_setup_matrices` to take a nontrivial amount of time.

Reply:
If it interferes with the GUI in the sense of locking it out til it is finished, I don't really mind.